### ISSUE TITLE
segfault in malloc_hook.fork on arm

### ISSUE THREAD

**Issue by yosefe** [openucx/ucx#1566]
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3733/label=hpc-arm-04,worker=2/console
without valgrind
```
13:54:17 [----------] 1 test from malloc_hook, where TypeParam = 
13:54:17 [ RUN      ] malloc_hook.fork
13:54:17 [     INFO ] Reduced heap free space to minimum: 9994544 after allocating 680000 bytes
13:54:17 [hpc-arm-04:39618:3] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x249c8a08)
13:54:17 [hpc-arm-04:39618:3] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x249c8a08)
13:54:17 [hpc-arm-04:39618:3] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x249c8a08)
13:54:17 [hpc-arm-04:39618:3] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x249c8a08)
15:15:45 Build timed out (after 120 minutes). Marking the build as failed.
15:15:45 Build was aborted
```

**Comment by shamisp** [comment]
have not seen this one

**Comment by alinask** [comment]
```
16:27:16 [----------] 1 test from malloc_hook, where TypeParam = 
16:27:16 [ RUN      ] malloc_hook.fork
16:27:16 [     INFO ] Reduced heap free space to minimum: 9997168 after allocating 4580000 bytes
16:27:16 [hpc-arm-04:13675:0] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x366e0400)
16:27:16 [hpc-arm-04:13675:0] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x366e0400)
16:27:16 [hpc-arm-04:13675:0] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x366e0400)
16:27:16 [hpc-arm-04:13675:0] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x366e0400)
```

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4183/label=hpc-arm-04,worker=2/console

**Comment by alinask** [comment]
A hang on the same test:

```
21:09:03 [----------] 1 test from malloc_hook, where TypeParam = 
21:09:03 [ RUN      ] malloc_hook.fork
21:09:16 [     INFO ] Reduced heap free space to minimum: 9990288 after allocating 85090000 bytes
21:09:16 [     INFO ] process 0/4: small mapped: 2093056, total mapped: 127942656, total unmapped: 127942656
21:09:16 [     INFO ] process 1/4: small mapped: 2093056, total mapped: 127942656, total unmapped: 127942656
21:09:16 [     INFO ] process 3/4: small mapped: 2093056, total mapped: 127942656, total unmapped: 127942656
23:32:15 
make: *** [test] Terminated
23:32:15 
```

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5651/label=vulcan-jenkins,worker=3/console

**Comment by yosefe** [comment]
fixed by 15af5ae9419137d445ab389028fac7def1ede7b3
